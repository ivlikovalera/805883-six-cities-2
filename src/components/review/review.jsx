import React from 'react';
import {PropTypes as pt} from 'prop-types';
import {connect} from 'react-redux';
import moment from 'moment';
import {getFetching} from './../../reducer/data/selector.js';

export const Review = (props) => {
  const {
    name,
    avatarUrl,
    rating,
    id,
    comment,
    date
  } = props;
  return <li className="reviews__item" id={id}>
    <div className="reviews__user user">
      <div className="reviews__avatar-wrapper user__avatar-wrapper">
        <img className="reviews__avatar user__avatar" src={avatarUrl} width="54" height="54" alt="Reviews avatar"/>
      </div>
      <span className="reviews__user-name">
        {name}
      </span>
    </div>
    <div className="reviews__info">
      <div className="reviews__rating rating">
        <div className="reviews__stars rating__stars">
          <span style={{width: `${rating / 5 * 100}%`}}></span>
          <span className="visually-hidden">{rating}</span>
        </div>
      </div>
      <p className="reviews__text">
        {comment}
      </p>
      <time className="reviews__time" dateTime={moment(date).format(`YYYY-MM-DD`)}>{moment(date).format(`MMMM YYYY`)}</time>
    </div>
  </li>;
};

Review.propTypes = {
  id: pt.number.isRequired,
  name: pt.string.isRequired,
  avatarUrl: pt.string.isRequired,
  rating: pt.number.isRequired,
  comment: pt.string.isRequired,
  date: pt.string.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isFetching: getFetching(state),
});

export default connect(mapStateToProps)(Review);
