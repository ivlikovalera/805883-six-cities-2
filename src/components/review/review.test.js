import React from 'react';
import renderer from 'react-test-renderer';
import {Review} from './review.jsx';

it(`Review correctly renders after relaunch`, () => {

  const reviewComponent = renderer
  .create(<Review
    id={0}
    name={``}
    avatarUrl={``}
    rating={0}
    comment={``}
    date={``}
  />)
  .toJSON();

  expect(reviewComponent).toMatchSnapshot();
});
