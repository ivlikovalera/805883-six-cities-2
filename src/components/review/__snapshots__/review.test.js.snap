// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review correctly renders after relaunch 1`] = `
<li
  className="reviews__item"
  id={0}
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src=""
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          0
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      
    </p>
    <time
      className="reviews__time"
      dateTime="Invalid date"
    >
      Invalid date
    </time>
  </div>
</li>
`;
